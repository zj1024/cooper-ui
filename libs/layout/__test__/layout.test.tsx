import * as React from 'react'
import * as renderer from 'react-test-renderer'
import Layout from '../index'

describe('- Layout component', () => {
  it(`Layout.Header has children 'Header'`, () => {
    const component = renderer.create(<Layout.Header>Header</Layout.Header>)
    let tree: any = component.toTree()
    expect(tree.props.children).toEqual('Header')
  })

  it(`Layout.Content has children 'Content'`, () => {
    const component = renderer.create(<Layout.Content>Content</Layout.Content>)
    let tree: any = component.toTree()
    expect(tree.props.children).toEqual('Content')
  })

  it(`Layout.Footer has children 'Footer'`, () => {
    const component = renderer.create(<Layout.Footer>Footer</Layout.Footer>)
    let tree: any = component.toTree()
    expect(tree.props.children).toEqual('Footer')
  })

  it(`Layout.Aside has children 'Aside'`, () => {
    const component = renderer.create(<Layout.Aside>Aside</Layout.Aside>)
    let tree: any = component.toTree()
    expect(tree.props.children).toEqual('Aside')
  })

  it(`Layout has child Aside will setClass 'coo-layout-has-aside'`, () => {
    const { Aside, Content } = Layout
    const component = renderer.create(
      <Layout>
        <Aside>aside</Aside>
        <Content>content</Content>
      </Layout>,
    )
    let tree: any = component.toJSON()
    expect(tree.props.className.includes('coo-layout-has-aside')).toBeTruthy()
  })
})
