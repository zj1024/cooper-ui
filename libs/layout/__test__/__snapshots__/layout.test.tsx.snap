// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`- Layout component Layout has child Aside will setClass 'coo-layout-has-aside' 1`] = `
<section
  className="coo-layout"
>
  <header
    className="coo-layout-header"
  >
    header
  </header>
  <section
    className="coo-layout coo-layout-has-aside"
  >
    <aside
      className="coo-layout-aside"
    >
      aside
    </aside>
    <main
      className="coo-layout-content"
    >
      content
    </main>
  </section>
  <footer
    className="coo-layout-footer"
  >
    footer
  </footer>
</section>
`;
